app/scripts/controllers/preferences.js

```diff
@@ -233,7 +252,11 @@ class PreferencesController {
       tokens.push(newEntry)
     }
 
-    this.store.updateState({ tokens })
+    const selectedAddress = this.store.getState().selectedAddress
+    const accountTokens = this.store.getState().accountTokens
+    const providerType = this.network.providerStore.getState().type
+    accountTokens[selectedAddress][providerType] = tokens
+    this.store.updateState({ accountTokens, tokens })
```

Comment:
Seems like there's an opportunity to de-duplicate some logic between here and `setAddress`.

Or at least, this function is getting a little big, we could probably tuck more of the token managing code into its own methods (eventually might want its own controller).